### PR TITLE
[connectors] Fix(Zendesk) - users not found

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Fix issues similar to [these](https://app.datadoghq.eu/logs?query=%40dd.service%3Aconnectors-worker%20%40dd.env%3Aprod%20%40hostname%3Aconnectors-worker-zendesk-deployment-%2A%20status%3Aerror%20%22TypeError%3A%20users.find%20is%20not%20a%20function%20%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&graphType=flamegraph&messageDisplay=inline&refresh_mode=sliding&sort=time&spanID=6000198897476506285&storage=hot&stream_sort=time%2Cdesc&viz=stream&from_ts=1732567949691&to_ts=1732571549691&live=true).
- The response most likely contains an object with error details instead of an array.
- Since this only occurred once, this will be logged and set aside for now.

## Risk

Low.

## Deploy Plan

- Deploy connectors.

### CODE REVIEW

File: connectors/src/connectors/zendesk/temporal/activities.ts [hunk 1]
```diff
@@ -427,7 +427,7 @@ export async function syncZendeskTicketBatchActivity({
     return { hasMore: false, nextLink: "" };
   }
 
-  const users = await zendeskApiClient.users.list();
+  const users = (await zendeskApiClient.users.list()) || [];
```
Comment (fontanierh) [inline connectors/src/connectors/zendesk/temporal/activities.ts:430]: can `.list()` return undefined ?

**Review by fontanierh — APPROVED** [review]

File: connectors/src/connectors/zendesk/lib/sync_ticket.ts [hunk 2]
```diff
@@ -175,7 +175,16 @@ ${metadata}
...
+    let author;
+    try {
+      author = users.find((user) => user.id === comment.author_id);
+    } catch (e) {
+      logger.warn(
+        { connectorId, e, ...loggerArgs },
+        "[Zendesk] Error finding the author of a comment."
+      );
+      author = null;
+    }
```
Comment (albandum) [inline connectors/src/connectors/zendesk/lib/sync_ticket.ts:187]:
```suggestion
   const author = users?.find((user) => user?.id === comment?.author_id) || null;
```

**Review by albandum — COMMENTED** [review]
I think we're not fond of trycatch here :P

File: connectors/src/connectors/zendesk/temporal/activities.ts (hunk 1 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/activities.ts:430]: not in the types, yes in the actual implementation

File: connectors/src/connectors/zendesk/lib/sync_ticket.ts (hunk 2 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/lib/sync_ticket.ts:187]: that's not really what we're trying to fix with our try catch, will safely ignore for now

Comment (fontanierh) [reply connectors/src/connectors/zendesk/lib/sync_ticket.ts:187]: Yeah actually not sure what this is supposed to do ?

Comment (fontanierh) [reply connectors/src/connectors/zendesk/lib/sync_ticket.ts:187]: if there's an error here then it's a typescript error no ? there's no async at all

**Pull request merged**